templates/module/default/hooks/windows-module.js

```diff
@@ -130,14 +127,23 @@ function runCMake(logger, cmake, projectDir, targetEnv, targetArch, targetGenera
 		originalTargetDir = null;
 	}
 
-	p = spawn(cmake,
-		[
+	var cmakeArgs =	[
 			'-G', generatorName,
 			'-DCMAKE_SYSTEM_NAME=' + targetPlatform,
 			'-DCMAKE_SYSTEM_VERSION=' + targetVersion,
 			'-DHAL_RENAME_AXWAYHAL=ON',
 			path.join(projectDir)
-		],
+		];
+
+	if (targetArch === 'ARM') {
+		cmakeArgs.push('-A', 'ARM');
+	} else if (targetArch === 'ARM64') {
+		cmakeArgs.push('-A', 'ARM64');
+	} else if (targetArch === 'x64') {
+		cmakeArgs.push('-A', 'x64');
+	}
```

Comment:
When generating the project I'm seeing the Win32 project get generated as a x64 project which then causes the build of the project to fail. Adding the below causes this the correct project type to be generated 
```suggestion
	} else if (targetArch === 'Win32') {
		cmakeArgs.push('-A', 'Win32');
	}
```

Reply:
Nice catch 👍 Fixed.